x-pack/legacy/plugins/apm/public/utils/formatters/__test__/datetime.test.ts

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import moment from 'moment-timezone';
+import { asRelativeDateTimeRange, asAbsoluteDateTime } from '../datetime';
+
+describe('date time formatters', () => {
+  describe('asRelativeDateTimeRange', () => {
+    const formatDateToTimezone = (
+      date: string,
+      timezone: string = 'Europe/Amsterdam'
+    ) =>
+      moment(date)
+        .tz(timezone)
+        .valueOf();
+
+    describe('years range', () => {
+      it('returns years range when difference is greater than 5 years', () => {
+        const start = formatDateToTimezone('2000-01-01 01:01:01');
+        const end = formatDateToTimezone('2010-01-01 01:01:01');
+        const dateRange = asRelativeDateTimeRange(start, end);
+        expect(dateRange).toEqual('2000 - 2010');
+      });
+      it('returns years range when difference is equal to 5 years', () => {
+        const start = formatDateToTimezone('2010-01-01 01:01:01');
+        const end = formatDateToTimezone('2015-01-01 01:01:01');
+        const dateRange = asRelativeDateTimeRange(start, end);
+        expect(dateRange).toEqual('2010 - 2015');
+      });
+    });
+    describe('months range', () => {
+      it('when difference is equal to 4 years ', () => {
+        const start = formatDateToTimezone('2010-01-01 01:01:01');
+        const end = formatDateToTimezone('2014-04-01 01:01:01');
+        const dateRange = asRelativeDateTimeRange(start, end);
+        expect(dateRange).toEqual('Jan 2010 - Apr 2014');
+      });
+      it('when difference is equal to 6 months ', () => {
+        const start = formatDateToTimezone('2019-01-01 01:01:01');
+        const end = formatDateToTimezone('2019-07-01 01:01:01');
+        const dateRange = asRelativeDateTimeRange(start, end);
+        expect(dateRange).toEqual('Jan 2019 - Jul 2019');
+      });
+    });
+    describe('days range', () => {
+      it('when difference is greater than 1 days', () => {
+        const start = formatDateToTimezone('2019-10-01 01:01:01');
+        const end = formatDateToTimezone('2019-10-05 01:01:01');
+        const dateRange = asRelativeDateTimeRange(start, end);
+        expect(dateRange).toEqual('Oct 1, 2019 - Oct 5, 2019');
+      });
+      it('when difference is equal to 1 days', () => {
+        const start = formatDateToTimezone('2019-10-01 01:01:01');
+        const end = formatDateToTimezone('2019-10-03 01:01:01');
+        const dateRange = asRelativeDateTimeRange(start, end);
+        expect(dateRange).toEqual('Oct 1, 2019 - Oct 3, 2019');
```

Comment:
I decided to only use the hours range when the difference is lower than 1 day, otherwise, the range would be too big (MMM D, YYYY, HH:mm - MMM D, YYYY, HH:mm)